src/wp-includes/blocks.php

```diff
@@ -760,11 +760,12 @@ function get_hooked_blocks( $name ) {
  * @since 6.4.0
  * @access private
  *
- * @param WP_Block_Template|array $context A block template, template part, or pattern that the blocks belong to.
+ * @param WP_Block_Template|array $context     A block template, template part, or pattern that the blocks belong to.
+ * @param array                   $block_hooks An array of blocks hooked to another given block.
  * @return callable A function that returns the serialized markup for the given block,
  *                  including the markup for any hooked blocks before it.
  */
-function make_before_block_visitor( $context ) {
+function make_before_block_visitor( $context, $block_hooks ) {
```

Comment:
We could change the argument order maybe? 🤔 `$block_hooks` is kinda essential for the whole thing, whereas `$context` is only needed for the filter.

Reply:
Also, `$hooked_blocks` maybe? See https://github.com/WordPress/wordpress-develop/pull/5399/files#r1350125889 😬

Reply:
I think I had some issues with variable shadowing or too close similarity with another variable. However, I will take another try to keep the `$hooked_blocks` as a param passed to the function.

Reply:
> I think I had some issues with variable shadowing or too close similarity with another variable. However, I will take another try to keep the `$hooked_blocks` as a param passed to the function.

Ah, didn't realize! Thank you, it'd be cool to keep it 😄 

> Sure, I can move the param to the first position. It's late in the process, but if we would get agreement from the release squad, we could also consider renaming the helper function to be more specific now it requires the list of hooked blocks.

You mean `make_before_block_visitor` (and `make_after_block_visitor`)? I’m struggling to come up with a good name for those 🤔 Arguably, they are still visitor factories — they just also accept a list of hooked blocks now…

Do you have anything specific in mind that should be reflected by their names?

Reply:
I addressed all the feedback with https://github.com/WordPress/wordpress-develop/pull/5399/commits/21efb9edb6f4ef1881cdb6bed8fa1c97c65ba0f6.

I don't have a clear vision how to rename these helpers so maybe it's fine to leave them as they are.

Reply:
Thanks for the discussion here, I am also in favor of the current approach with `$hooked_blocks` as the first parameter, so LGTM.

Note that the function is marked private, that would still give us some additional flexibility for the future if we wanted to change things.